examples/systemcalls/systemcalls.c

```diff
@@ -7,15 +14,22 @@
  *   either in invocation of the system() call, or if a non-zero return
  *   value was returned by the command issued in @param cmd.
 */
-bool do_system(const char *cmd)
-{
 
-/*
- * TODO  add your code here
- *  Call the system() function with the command set in the cmd
- *   and return a boolean true if the system() call completed with success
- *   or false() if it returned a failure
-*/
+#define MAX_OUTPUT_SIZE 1024
+
+bool do_system(const char *cmd) {
+    // Create a pipe to capture the command's output
+    int status = system(cmd);
+
+    if(-1 == status){
+	 //Log error
+	  return false;
+    }
+	
+    //What is this???
+    if(!WIFEXITED(status)){
+	    return false;
```

Comment:
@Suraj-Ajjampur will these checks ensure you return false if the `cmd` passed to `system` fails with any return code other than 0?

Reply:
Edited to obtain the exit status status of the child process 
```
    if (WIFEXITED(status)) {
        // Child process exited normally
        return (WEXITSTATUS(status) == 0);
    } 
    else 
    {
            // Child process did not exit normally
            return false;
    }
```